### PR TITLE
Feat 4063 fix safari bug and campaign form

### PR DESCRIPTION
**Pull request by phpgeekfr**

### CODE REVIEW

**Review by fritzschoff — APPROVED** [review]

**Review by Mussuranae — APPROVED** [review]

File: libs/movie/src/lib/movie/form/shell/shell.component.ts [hunk 1]
```diff
@@ -172,8 +172,10 @@ export class MovieFormShellComponent implements TunnelRoot, OnInit, AfterViewIni
 
   ngOnDestroy() {
     this.sub?.unsubscribe();
-    this.getForm('movie').reset()
-    this.getForm('campaign').reset()
+    this.getForm('movie').reset();
+    if (this.configs[name]) {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/shell/shell.component.ts:176]: Where is "name" defined ?

File: libs/movie/src/lib/movie/dashboard/shell/shell.component.html [hunk 2]
```diff
@@ -1,5 +1,5 @@
 <section fxLayout="column">
-  <header fxLayout="row" fxLayout.lt-md="column" fxFlex>
+  <header fxLayout="row" fxLayout.lt-md="column" fxFlex="100%">
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/dashboard/shell/shell.component.html:2]: What is the issue ?

File: libs/movie/src/lib/movie/form/shell/shell.component.ts (hunk 1 above)
Comment (phpgeekfr) [reply libs/movie/src/lib/movie/form/shell/shell.component.ts]: typo, fixed

File: libs/movie/src/lib/movie/dashboard/shell/shell.component.html (hunk 2 above)
Comment (phpgeekfr) [reply libs/movie/src/lib/movie/dashboard/shell/shell.component.html:2]: image

File: libs/movie/src/lib/movie/form/shell/shell.component.ts [hunk 3]
```diff
@@ -173,7 +173,7 @@ export class MovieFormShellComponent implements TunnelRoot, OnInit, AfterViewIni
   ngOnDestroy() {
     this.sub?.unsubscribe();
     this.getForm('movie').reset();
-    if (this.configs[name]) {
+    if (this.configs['campaign']) {
```
Comment (GrandSchtroumpf) [inline libs/movie/src/lib/movie/form/shell/shell.component.ts:176]: Prefer `this.getForm('campaign')?.reset()` then. But I don't understand why you need that.

Comment (phpgeekfr) [reply libs/movie/src/lib/movie/form/shell/shell.component.ts]: I had to add `?` also to 
``` getForm<K extends keyof ShellConfig>(name: K): ShellConfig[K]['form'] {
    return this.configs[name]?.form;
  }
```
but ok

File: libs/movie/src/lib/movie/dashboard/shell/shell.component.html (hunk 2 above)
Comment (phpgeekfr) [reply libs/movie/src/lib/movie/dashboard/shell/shell.component.html:2]: @GrandSchtroumpf Waiting for your test before removing this

**Pull request merged**